### PR TITLE
Feature/9119 booster notification/9212 rule check

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR adds an API to check the certificates with rules for the case of booster notifications.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9212

### CODE REVIEW

File: src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess/Model/DigitalCovidCertificateWithHeader.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+public struct DigitalCovidCertificateWithHeader: Codable, Equatable {
+    let header: CBORWebTokenHeader
+    let certificate: DigitalCovidCertificate
+
+    static func fake(
```
Comment (marcussc) [inline src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess/Model/DigitalCovidCertificateWithHeader.swift:11]: Maybe wrap this into `#ifdef ` so it does not get used in prod somewhere?

File: src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/RuleValidation/ValidationRulesAccess.swift [hunk 2]
```diff
@@ -64,4 +64,110 @@ public struct ValidationRulesAccess: ValidationRulesAccessing {
...
+        }
+
+        let recentVaccinationCertificate = recentVaccinationCertificateWithHeader.certificate
+
+        var recoveryEntries: [RecoveryEntry]?
+        if let recoveryEntry = certificates.recentRecoveryCertificate?.certificate.recoveryEntries?[0] {
+            recoveryEntries = [recoveryEntry]
+        }
+
+        // Prepare payload: the payload shall be the JSON representation of the most recent vaccination certificate.
```
Comment (naveeddotio) [inline src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/RuleValidation/ValidationRulesAccess.swift:87]: 👍

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess/Model/DigitalCovidCertificateWithHeader.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess/Model/DigitalCovidCertificateWithHeader.swift:13]: Done. Also added them to other mock, fake stuff in the lib.

**Review by marcussc — APPROVED** [review]
LGTM especially I liked the Comments.

**Pull request merged**